be1-go/hub/base_channel_test.go

```diff
@@ -0,0 +1,39 @@
+package hub
+
+import (
+	"bytes"
+	"student20_pop/message"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestBaseChannel_RollCallOrder(t *testing.T) {
+	// Create the channel
+	channel := createBaseChannel(nil, "Channel0")
+
+	// Create the messages
+	messageNumber := 10
+	messages := make([]message.Message, 0, messageNumber)
+	for i := 0; i < messageNumber; i++ {
+		// Create a new message containing only an id
+		message := message.Message{MessageID: []byte{byte(i)}}
+		messages = append(messages, message)
+
+		// Store the message in the inbox
+		channel.inbox.storeMessage(message)
+
+		// Wait before storing a new message
+		time.Sleep(time.Microsecond)
```

Comment:
I believe we need this to get a unique timestamp for each message in the test?

Reply:
Yes! Maybe it would be better to just increment the timestamp